ethsigner/signing-api/src/main/java/tech/pegasys/ethsigner/core/signing/SingleTransactionSignerFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.signing;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+public class SingleTransactionSignerFactory implements TransactionSignerFactory {
+
+  private TransactionSigner signer;
+
+  public SingleTransactionSignerFactory(final TransactionSigner signer) {
+    if (signer == null) {
+      throw new IllegalArgumentException(
+          "SingleTransactionSignerFactory requires a non-null TransactionSigner");
+    }
+    this.signer = signer;
+  }
+
+  @Override
+  public Optional<TransactionSigner> getSigner(final String address) {
+    if (signer.getAddress() != null && signer.getAddress().equalsIgnoreCase(address)) {
```

Comment:
Is a signer allowed to have a null address?

Reply:
Well, TxSigner is an interface so, in theory, someone could implement a version of it that returned an empty address. I believe it shouldn't.

If it does though, we handle it and return Optional.empty().